src/components/Chatbox/Input.tsx

```diff
@@ -0,0 +1,241 @@
+import clsx from 'clsx';
+
+import React, { FC, ReactElement, RefObject, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Mention,
+  MentionsInput,
+  OnChangeHandlerFunc,
+  SuggestionDataItem,
+} from 'react-mentions';
+
+import { Typography } from '@material-ui/core';
+import Box from '@material-ui/core/Box';
+import IconButton from '@material-ui/core/IconButton';
+import { makeStyles } from '@material-ui/core/styles';
+import SendIcon from '@material-ui/icons/Send';
+
+import { MessageBodyType } from '@graasp/query-client/dist/src/types';
+import { CHATBOX } from '@graasp/translations';
+
+import {
+  charCounterCypress,
+  inputTextFieldCypress,
+  inputTextFieldTextAreaCypress,
+  sendButtonCypress,
+} from '../../config/selectors';
+import {
+  ALL_MEMBERS_DISPLAY,
+  ALL_MEMBERS_ID,
+  HARD_MAX_MESSAGE_LENGTH,
+} from '../../constants';
+import { useCurrentMemberContext } from '../../context/CurrentMemberContext';
+import { useMessagesContext } from '../../context/MessagesContext';
+import { getAllMentions, normalizeMentions } from '../../utils/mentions';
+
+type Props = {
+  id?: string;
+  inputRef: RefObject<HTMLTextAreaElement>;
+  placeholder?: string;
+  textInput: string;
+  setTextInput: (newText: string) => void;
+  sendMessageFunction?: (body: MessageBodyType) => void;
+};
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    width: '100%',
+  },
+  textLength: {
+    whiteSpace: 'pre',
+    paddingLeft: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    color: 'gray',
+  },
+  textTooLong: {
+    color: 'red !important',
+  },
+}));
+
+const inputStyle = {
+  width: '100%',
+  // mentions
+  control: {
+    // fontFamily: 'monospace',
+    minHeight: 63,
+  },
+  input: {
+    padding: 9,
+    border: '1px solid silver',
+    width: '100%',
+    overflow: 'auto',
+    height: 70,
+    borderRadius: '4px',
+  },
+  highlighter: {
+    padding: 9,
```

Comment:
It's supposed to be px? what about theme.spacing?

Reply:
This value works for the mentions, if we change it to something different it will change the layout. I agree it is not the best but i think we don't have the choice ...